src/control/lib/control/pool.go

```diff
@@ -372,7 +372,35 @@ const (
 )
 
 func (prs PoolRebuildState) String() string {
-	return [...]string{"idle", "done", "busy"}[prs]
+	return strings.ToLower(mgmtpb.PoolRebuildStatus_State_name[int32(prs)])
+}
+
+func (prs PoolRebuildState) MarshalJSON() ([]byte, error) {
+	stateStr, ok := mgmtpb.PoolRebuildStatus_State_name[int32(prs)]
+	if !ok {
+		return nil, errors.Errorf("invalid rebuild state %d", prs)
+	}
+	return []byte(`"` + strings.ToLower(stateStr) + `"`), nil
+}
+
+func (prs *PoolRebuildState) UnmarshalJSON(data []byte) error {
+	stateStr := strings.ToUpper(string(data))
+	state, ok := mgmtpb.PoolRebuildStatus_State_value[stateStr]
+	if !ok {
+		// Try converting the string to an int32, to handle the
+		// conversion from protobuf message using convert.Types().
+		if si, err := strconv.ParseInt(stateStr, 0, 32); err == nil {
+			if _, ok = mgmtpb.PoolRebuildStatus_State_name[int32(si)]; !ok {
+				return errors.Errorf("invalid rebuild state %q", stateStr)
+			}
+			state = int32(si)
+		} else {
```

Comment:
if you switch the logic then you can avoid the else clause and dedent the type assertion

Reply:
Hmm, but in order to assign to si and err from ParseInt I'd have to predeclare them, right?

Reply:
Yeah, I think it would be better to do something more like:
```
si, err := ParseInt(...)
if err != nil {
        return errors.Errorf("invalid rebuild state %q", stateStr)
}
// process si...
```

Reply:
Errr... Yes. You're both right, thanks. I had some kind of mental block that was keeping me from seeing that. Fixed.